### PR TITLE
[VC02-3555]-testtool update

### PR DESCRIPTION
**Pull request by EwaGrela**

### CODE REVIEW

File: supplier_api_tester/supplier_api_tester/utils/adapters.py [hunk 1]
```diff
@@ -8,6 +11,29 @@
...
+    try:
+        return base64.b64encode(base64.b64decode(item)).decode() == item
+    except binascii.Error:
+        return False
+
+def booking_pdf_helper(booking: Booking, raw_response:Request):
+    if booking.barcode_position == 'order':
+        if not check_base64(booking.barcode):
+            raise FailedTest(
+                message = "Invalid barcode field encoding",
```
Comment (trojkat) [inline supplier_api_tester/supplier_api_tester/utils/adapters.py:24]: Maybe something like `Error while decoding (base64) PDF voucher for the order` would be more descriptive to developers on where the issue is.

File: supplier_api_tester/supplier_api_tester/utils/adapters.py [hunk 2]
```diff
@@ -178,7 +204,10 @@ def get_booking(raw_response: Request, response) -> Booking:
             response=raw_response,
         )
 
-    if booking.barcode_format not in ('QRCODE', 'CODE128', 'CODE39', 'ITF', 'DATAMATRIX', 'EAN13'):
+    if booking.barcode_format == 'PDF':
```
Comment (trojkat) [inline supplier_api_tester/supplier_api_tester/utils/adapters.py:207]: Place this check after the one that checks the barcode_format code.

File: supplier_api_tester/supplier_api_tester/models.py [hunk 3]
```diff
@@ -46,6 +46,9 @@ class Product:
     name: str
     use_timeslots: bool
     description: Optional[str]
+    is_refundable: bool
+    cutoff_time: int
+    ticket_voucher_type: Optional[str]
```
Comment (trojkat) [inline supplier_api_tester/supplier_api_tester/models.py:51]: This field is not a part of the API spec.

File: supplier_server_mock/supplier_server/utils.py [hunk 4]
```diff
@@ -76,14 +76,15 @@ def get_availability(product_id: str, day: date):
     }
 
 
-def encode_reservation_id(expires_at: datetime, tickets: list) -> str:
+def encode_reservation_id(expires_at: datetime, tickets: list, product_id: str) -> str:
     variants_quantity_map = {ticket['variant_id']: ticket['quantity'] for ticket in tickets}
-    json_content = json.dumps([expires_at.isoformat(), variants_quantity_map])
+    json_content = json.dumps([expires_at.isoformat(), variants_quantity_map, {"product_id": product_id}])
```
Comment (trojkat) [inline supplier_server_mock/supplier_server/utils.py:81]: Why dict and not just integer?

File: supplier_server_mock/supplier_server/app.py [hunk 5]
```diff
@@ -152,9 +153,10 @@ def booking():
     for variant_id, quantity in variant_quantity_map.items():
         barcodes = [str(utils.str_to_int(f'{reservation_id}{variant_id}{i}', 10)) for i in range(quantity)]
```
Comment (trojkat) [inline supplier_server_mock/supplier_server/app.py:154]: For PDF-s we still don't return base64 encoded content.

**Comment by trojkat** [comment]
Fo the mock server we also need to release a new version and update the changelog.

File: supplier_server_mock/supplier_server/app.py [hunk 6]
```diff
@@ -129,32 +130,34 @@ def reservation(product_id: str):
...
+
         raise exceptions.BadRequest(3002, 'Incorrect reservation ID', 'Given reservation ID is incorrect')
     now = arrow.utcnow().datetime
     if now > expires_at:
         minutes_ago = round((now - expires_at).seconds / 60)
         raise exceptions.BadRequest(3001, 'Reservation expired', f'Your reservation has expired {minutes_ago} minutes ago')
     tickets = {}
     for variant_id, quantity in variant_quantity_map.items():
-        barcodes = [str(utils.str_to_int(f'{reservation_id}{variant_id}{i}', 10)) for i in range(quantity)]
+        barcodes = [utils.encode_barcode(f'{reservation_id}{variant_id}{i}') for i in range(quantity)]
```
Comment (trojkat) [inline supplier_server_mock/supplier_server/app.py:155]: An `if` is missing here as we want to use base64 only for PDF products.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+# 1.4.1
+- Updating mock server with support for PDF type
+- Updating test-tool with testing validity of tickets for PDF type
```
Comment (trojkat) [inline CHANGELOG.md:5]: This is a changelog for the specification so no changes here.

Comment (EwaGrela) [reply CHANGELOG.md]: revoked

**Review by trojkat — APPROVED** [review]
:ok_hand:

**Pull request merged**